src/storage/storage.rs

```diff
@@ -87,12 +93,106 @@ pub trait Storage: Sync + Send {
     }
 }
 
+/// Starknet hash type.
+/// Encapsulates the result of hash functions and provides conversion functions to Cairo VM
+/// and Starknet API types for convenience.
+#[derive(Debug, Clone, Hash, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(transparent)]
+pub struct Hash([u8; 32]);
+
+impl Hash {
+    pub fn empty() -> Self {
+        Self::from_bytes_be(EMPTY_NODE_HASH)
+    }
+
+    pub fn from_bytes_be(bytes: [u8; 32]) -> Self {
+        Self(bytes)
+    }
+
+    pub fn from_bytes_be_slice(bytes: &[u8]) -> Self {
```

Comment:
This could use some documentation about behavior WRT input slice length. It looks to me like the only sane way to use it would be to pass an array of exactly 32 bytes, so maybe it needs an assertion about this?

Reply:
Yeah it could use more doc. Only sane way is if the input is 32 bytes indeed, I implemented it in a way that it supports smaller (add padding) and larger inputs (ignore everything after byte 32).